### PR TITLE
fix: add an input for percentile in metrics

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6773 

### Description:

Custom metric for percentile didn't have anywhere to choose the percentile. This adds an input box to choose what you want the percentile to be and plumbs that through. 

I hadn't interacted with these things much -- if anyone sees something I didn't think of or should have done differently, I'm very open to changing this.

### CODE REVIEW

File: packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx [hunk 1]
```diff
@@ -127,45 +136,50 @@ export const CustomMetricModal = () => {
...
+                percentile,
             });
-            showToastSuccess({
-                title: 'Custom metric added successfully',
-            });
-        }
-        toggleModal();
-    });
+
+            console.log(data);
```
Comment (joaoviana) [inline packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx:154]: you can remove this 👍

**Comment by joaoviana** [comment]
@all-contributors add @magnew bug

File: packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx [hunk 2]
```diff
@@ -201,6 +215,14 @@ export const CustomMetricModal = () => {
                         placeholder="Enter custom metric label"
                         {...form.getInputProps('customMetricLabel')}
                     />
+                    {customMetricType === MetricType.PERCENTILE && (
+                        <NumberInput
+                            w={100}
+                            required
```
Comment (joaoviana) [inline packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx:221]: I would also add: 
```
   max={100}
   min={0}
```

File: packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts [hunk 3]
```diff
@@ -95,6 +97,11 @@ export const prepareCustomMetricData = ({
             ? { round: dimension.round }
             : defaultRound;
 
+    const percentile =
+        shouldCopyFormatting && dimension.percentile
+            ? dimension.percentile
+            : metricPercentile;
```
Comment (joaoviana) [inline packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts:103]: wouldn't this always send `metricPercentile` `50` if it was non-percentile?

**Review by joaoviana — CHANGES_REQUESTED** [review]
Tested the following: 
* Creating/editing chart: added custom metric; edited; added another percentile type & non-percentile custom metric;
* Added chart to dashboard; Explored, all good

Left some changes/suggestions!

File: packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx:154]: Oops. Will do.

File: packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx:221]: Good call.

File: packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts (hunk 3 above)
Comment (magnew) [reply packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts:103]: Hm, yea. It should be benign, but good catch. Will change.

**Review by joaoviana — APPROVED** [review]

**Pull request merged**